app/presenters/firm_profile_presenter.rb

```diff
@@ -1,7 +1,55 @@
-class FirmProfilePresenter < SimpleDelegator
-  def type_of_advice_method
-    return :face_to_face if in_person_advice_methods.present?
+class FirmProfilePresenter
+  include ApplicationHelper
 
-    :remote
+  def initialize(json)
+    @json = json
+  end
+
+  def firm
+    return unless advisers_hits.first&.fetch('firm')
+
+    hit = advisers_hits.first['firm']
+    Results::FirmPresenter.new(hit).tap do |firm|
+      firm.offices = offices
+      firm.advisers = advisers
+      firm.closest_adviser_distance = closest_adviser_distance
+    end
+  end
+
+  private
+
+  attr_reader :json
+
+  def advisers_hits
+    json[:advisers]['hits']
+  end
+
+  def offices_hits
+    json[:offices]['hits']
+  end
+
+  def offices
+    offices_hits.map do |hit|
+      Results::OfficePresenter.new(hit)
+    end
+  end
+
+  def advisers
+    advisers_hits.map do |hit|
+      Results::AdviserPresenter.new(hit.except('firm')).tap do |adviser|
+        adviser.distance = adviser_distance(hit)
+      end
+    end
+  end
+
+  def closest_adviser_distance
+    adviser_distance(advisers_hits.first)
+  end
+
+  def adviser_distance(hit)
+    return unless hit['_rankingInfo']
+
+    distance_in_meters = hit['_rankingInfo']['geoDistance']
```

Comment:
should some of these strings be constants?

Reply:
I don't know. IMHO it doesn't look like to me that this should be constants because then we will have constants for every node group of the json and also the node attributes. Is there a reason that you think that all of them should be constants and how will that improve?

Reply:
Having looked through more of the pr, I can see using constants wouldn't necessarily add anything.